authentication/account/worker/user_deactivation_worker.go

```diff
@@ -60,14 +61,32 @@ func (w userDeactivationWorker) deactivateUsers() {
 		// deactivating the user on OSO and then call back `auth` service (on its `/namedusers/:username/deactivate` endpoint)
 		// which will handle the deactivation on the OSIO platform
 		err = w.App.OSOSubscriptionService().DeactivateUser(w.Ctx, identity.Username)
-		metric.RecordUserDeactivationTrigger(err == nil)
 		if err != nil {
-			// We will just log the error and continue
-			log.Error(nil, map[string]interface{}{
-				"err":      err,
-				"username": identity.Username,
-			}, "error while deactivating user")
+			if _, ok := err.(autherrors.NotFoundError); ok {
+				// deactivate user directly
+				_, err := w.App.UserService().DeactivateUser(w.Ctx, identity.Username)
+				if err != nil {
+					metric.RecordUserDeactivationTrigger(false)
+					log.Error(nil, map[string]interface{}{
+						"err":      err,
+						"username": identity.Username,
+					}, "error during deactivating user")
+				} else {
+					metric.RecordUserDeactivationTrigger(true)
+					log.Info(nil, map[string]interface{}{
+						"username": identity.Username,
+					}, "user deactivation is successful")
+				}
+			} else {
+				// We will just log the error and continue
+				metric.RecordUserDeactivationTrigger(false)
```

Comment:
you can move this call to line 66 and remove the one on line 69

Reply:
make some changes, plz check commit [339a774](https://github.com/fabric8-services/fabric8-auth/pull/862/commits/339a7742b3b98a392d31941fcffe8a832ef595f2).  userService.DeactivateUser() takes care of recording metrics so no need to record metrics in case 404 error.

Reply:
good point, indeed ;)